### PR TITLE
SRCH-4119 Dockerize i14y development

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Add a Dockerfile and the configuration to the i14y.
- i14y service will be created and run using docker-compose.yml in search-services. Refer to https://github.com/GSA/search-services/pull/10
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM ruby:3.0.6
+WORKDIR /usr/src/app
+EXPOSE 3000
```
Comment (ddzz) [inline Dockerfile:3]: Should we expose and use a port other than the default `3000`? Same question for the ASIS PR.

Comment (ddzz) [reply Dockerfile:3]: Yes, I think it's better to use unique ports like in the docker-compose file to lower the risk of port conflicts.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+RUN apt install -y curl
+
+COPY Gemfile* /usr/src/app/
```
Comment (ddzz) [inline Dockerfile:7]: Is there a reason we copy only `Gemfile*` here and then copy everything in L11?

**Review by ddzz — APPROVED** [review]
LGTM

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ENV OPENSSL_CONF /etc/ssl/
+
+RUN apt install -y curl
+
+COPY Gemfile* /usr/src/app/
+ENV BUNDLE_PATH /gems
+RUN bundle install
+
+COPY . /usr/src/app/
+CMD ["rails", "server", "-b", "0.0.0.0"]
```
Comment (ddzz) [inline Dockerfile:14]: Can you add a `bundle exec` here?

Comment (krbhavith) [reply Dockerfile]: Added

**Review by ddzz — APPROVED** [review]

**Review by ddzz — DISMISSED** [review]

File: config/application.rb [hunk 4]
```diff
@@ -40,5 +40,6 @@ class Application < Rails::Application
     # Middleware like session, flash, cookies can be added back manually.
     # Skip views, helpers and assets when generating a new resource.
     config.api_only = true
+    config.hosts         << "i14y"
```
[the remark this answers is not in the document]
Comment (krbhavith) [reply config/application.rb:43]: I think it's a good idea to differentiate with local environment, implemented same for asis. I have updated env variables in `docker.compose.yml`

File: config/environments/test.rb [hunk 5]
```diff
@@ -46,4 +46,5 @@
 
   # Annotate rendered view with file names.
   # config.action_view.annotate_rendered_view_with_filenames = true
+  config.hosts << "www.example.com"
```
Comment (ddzz) [inline config/environments/test.rb:49]:
```suggestion
  config.hosts << "www.example.com" if ENV["DOCKER"]
```

**Review by ddzz — APPROVED** [review]
LGTM

**Pull request merged**